addons/stock_account/tests/test_account_move.py

```diff
@@ -363,6 +363,41 @@ def test_cogs_account_branch_company(self):
 
         self.assertEqual(bill.invoice_line_ids.account_id, test_account)
 
+    def test_no_currency_rate_on_cogs(self):
+        """Check that when confirming a invoice with no date and a different rate than the one
+        of the current date, no currency rate is applied to the cogs.
+        """
+        self.other_currency.rate_ids.sorted()[0].rate = 2
+
+        invoice = self.env['account.move'].with_context(default_move_type='out_invoice').create({
+            'partner_id': self.partner_a.id,
+            'currency_id': self.other_currency.id,
+            'invoice_currency_rate': 10,
+            'invoice_line_ids': [
+                Command.create({
+                    'product_id': self.product_A.id,
+                    'price_unit': 200,
+                    'tax_ids': [],
+                }),
+            ],
+        })
```

Comment:
We recently made a new helpers for creating moves in accounting tests. Can you try this and see if it works?
```suggestion
        invoice = self._create_invoice_one_line(
            currency_id=self.other_currency,
            invoice_currency_rate=10,
            product_id=self.product_A,
            price_unit=200,
        )

Reply:
Thanks for the review ! from what I found when trying, the helper does not allow to reproduce the issue, as it uses _create_invoice() which sets a date on the invoice from it's creation